src/test/python/parser/opentrackio_receiver.py

```diff
@@ -0,0 +1,254 @@
+	
+	if payloadformat == PayloadFormat.CBOR:
+		try:
+			opentrackiolib.parse_cbor(payload)
+		except OpenTrackIOException as e:
+			print(e)
+			return False
+	elif payloadformat == PayloadFormat.JSON:
+		try:
+			opentrackiolib.parse_json(payload)
+		except OpenTrackIOException as e:
+			print(e)
+			return False				
+	
+	if not timesource:
+		timesource = opentrackiolib.get_time_source()
+		init_time_source()
+	
+	ref_timestamp = get_local_timestamp()
+	ref_delta = ref_timestamp - opentrackiolib.get_sample_time(TimeFormat.SECONDS)
+	
+	try:
+		if VERBOSE:
+			print(f"OpenTrackIO packet:\n")
+			print(f"  Timecode: {opentrackiolib.get_timecode()}")
+			print(f"  Time source: {opentrackiolib.get_time_source().name}")
+			print(f"  Local time: {ref_timestamp}")
+			print(f"  Sample time: {opentrackiolib.get_sample_time(TimeFormat.SECONDS)}")
+			print(f"  Delta: {ref_delta}")
+			print(f"  Translation: {opentrackiolib.get_camera_translations()}")
+			print(f"  Rotation: {opentrackiolib.get_camera_rotations()}")
+	except OpenTrackIOException as e:
+		print(e)
+
+def get_local_timestamp():
+	if timesource == TimeSource.NTP:
+		timestamp = time.time() + ntpoffset
+		#print(f"Local time: {timestamp}")
+		return timestamp
+
+def main():
+	global VERBOSE, opentrackiolib
+	
+	parser = argparse.ArgumentParser(description='OpenTrackIO protocol receiver')
+	parser.add_argument('-n', '--source', type=int, help='The Source Number (1-200) to listen for.', default=None)
+	parser.add_argument('-p', '--port', type=int, help='The port number to listen on. Default: 55555.', default=None)
+	parser.add_argument('-s', '--schema', help='The schema (JSON) input file. Default: opentrackio_schema.json', default='opentrackio_schema.json')
+	parser.add_argument('-v', '--verbose', help='Make script more verbose', action='store_true')
+	args = parser.parse_args()
+	
+	schematext = ''
+	schemapath = None
+	source_number = None
+	multicast_port = MULTICAST_PORT
+
+	if (args.schema):
+		if os.path.exists(args.schema):
+			schemapath = args.schema
+			if (schemapath != None):        
+				with open(schemapath, 'r') as fd:
+					if VERBOSE:
+						print("Reading OpenTrackIO schema file: {0}".format(schemapath))
+					lines = fd.readlines()
+					for line in lines:
+						schematext = schematext + line		
+	
+	if (args.source):
+		if not (1 <= args.source <= 200):
+			print("Error: Source Number must be between 1 and 200.")
+			exit(-1)
+		else:
+			source_number = args.source	
+	
+	if (args.port):
+		if not (49152 <= args.port <= 65535):
+			print("Error: port number must be between 49153 and 65535.")
+			exit(-1)
+		else:
+			multicast_port = args.port	
+			
+	if (args.verbose):
+		VERBOSE = True
+	
+	if not source_number or not schematext:
+		print("Usage: python3 opentrackio_receiver.py --source=[1-200] --schema=opentrackio_schema.json --verbose")
+		exit(-1)		
+	
+	opentrackiolib = OpenTrackIOProtocol(schematext, False)	
+
+	multicast_group = f"235.135.1.{source_number}"
```

Comment:
The multicast group (and others?) could go into _lib and be referenced by both sender and receiver?